### PR TITLE
Fix  checking ammID field of account

### PR DESCRIPTION
**Pull request by oleks-rip**
## High Level Overview of Change

This fix add check for sfAMMID field in amm_info handler. 

### Context of Change

Without this check debug version of rippled crash on assert in [ledger::read()](https://github.com/XRPLF/rippled/blob/9d58f11a60881b3edb5de80cae52d8b758ae30a8/src/xrpld/app/ledger/Ledger.cpp#L445)
Bug was found during automation testing. 

### Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)

### API Impact

- [x] Public API:  amm_info will not crash debug version of the rippled

### CODE REVIEW

File: src/test/rpc/AMMInfo_test.cpp [hunk 1]
```diff
@@ -316,13 +316,45 @@ class AMMInfo_test : public jtx::AMMTestBase
         });
     }
 
+    void
+    testInvalidAmmField()
+    {
+        using namespace jtx;
+        testcase("Invalid amm field");
+
```
Comment (gregtatcam) [inline src/test/rpc/AMMInfo_test.cpp:324]: Consider refactoring to simplify and for consistency with other tests:

        testAMM([&](AMM& amm, Env&) {
            auto const resp = amm.ammRpcInfo(
                std::nullopt,
                jss::validated.c_str(),
                std::nullopt,
                std::nullopt,
                gw);
            BEAST_EXPECT(resp.isMember("error") && resp["error"] == "actNotFound");
        });

**Review by gregtatcam — APPROVED** [review]
👍 LGTM

**Comment by shawnxie999** [comment]
Unrelated to this change, but on line 162, an non existent AMM object is returned with rpcACT_NOT_FOUND error. Doesn't sound right, should we fix it?

```
        auto const amm = ledger->read(ammKeylet);
        if (!amm)
            return Unexpected(rpcACT_NOT_FOUND);
```

File: src/test/rpc/AMMInfo_test.cpp [hunk 2]
```diff
@@ -316,13 +316,32 @@ class AMMInfo_test : public jtx::AMMTestBase
...
+        testAMM([&](AMM& amm, Env&) {
+            auto const resp = amm.ammRpcInfo(
+                std::nullopt,
+                jss::validated.c_str(),
+                std::nullopt,
+                std::nullopt,
+                gw);
+            BEAST_EXPECT(
+                resp.isMember("error") && resp["error"] == "actNotFound");
+        });
```
Comment (shawnxie999) [inline src/test/rpc/AMMInfo_test.cpp:334]: nit: the test could be moved into `testErrors()` IMO

**Comment by gregtatcam** [comment]
> Unrelated to this change, but on line 162, an non existent AMM object is returned with rpcACT_NOT_FOUND error. Doesn't sound right, should we fix it?
> 
> ```
>         auto const amm = ledger->read(ammKeylet);
>         if (!amm)
>             return Unexpected(rpcACT_NOT_FOUND);
> ```

It's half correct since AMM might be fetched by amm account. But correct, `rpcOBJECT_NOT_FOUND` is probably more appropriate.

**Comment by shawnxie999** [comment]
> > Unrelated to this change, but on line 162, an non existent AMM object is returned with rpcACT_NOT_FOUND error. Doesn't sound right, should we fix it?
> > ```
> >         auto const amm = ledger->read(ammKeylet);
> >         if (!amm)
> >             return Unexpected(rpcACT_NOT_FOUND);
> > ```
> 
> It's half correct since AMM might be fetched by amm account. But correct, `rpcOBJECT_NOT_FOUND` is probably more appropriate.

although i believe we  may need a new API version for this change

**Review by godexsoft — APPROVED** [review]

**Review by intelliot — APPROVED** [review]

**Comment by oleks-rip** [comment]
Ready to be merged.

**Pull request merged**